configstack/running_module.go

```diff
@@ -114,10 +114,12 @@ func runModulesWithHandler(modules []*TerraformModule, handler ModuleHandler, or
 	}
 
 	if len(modules) != 0 {
-		if modules[0].TerragruntOptions.NbWorkers <= 0 {
-			modules[0].TerragruntOptions.NbWorkers = len(runningModules)
+		options := modules[0].TerragruntOptions
+		// We don't gain anything by running on more workers than there are modules
+		if options.NbWorkers <= 0 || options.NbWorkers > len(runningModules) {
```

Comment:
Couldn't you do `options.NbWorkers = Max(options.NbWorkers, len(runningModules))`?

Reply:
Good idea, I thought about it, but if `options.NbWorkers = 10` and we have only 4 modules, we would still have 10 workers for 4 modules. I tried to come up a solution with two min/max statements, but the condition translates into 
`nbWorkers != 0 && nbWorkers <= len(modules)` which I could not well represent as min/max statements.